server/src/main/java/org/elasticsearch/action/ingest/SimulateProcessorResult.java

```diff
@@ -159,7 +160,11 @@ public SimulateProcessorResult(String type, String processorTag, String descript
      * Read from a stream.
      */
     SimulateProcessorResult(StreamInput in) throws IOException {
-        this.processorTag = in.readString();
+        if (in.getVersion().onOrAfter(Version.V_8_7_0)) {
+            this.processorTag = in.readOptionalString();
+        } else {
+            this.processorTag = in.readString();
+        }
```

Comment:
This is a little strange, did we not serialize this before? It seems like we should have tests for this in `SimulateProcessorResultTests`, what do you think?

Reply:
Yeah it's definitely a bug, and I'm not sure we were ever even using this serialization code before. @joegallo and I were discussing whether there's even any value in making this check conditional. The only value is in a mixed cluster where the ingest node happens to be an older one and we forward a simulate request to it... but in that case it'll throw a NullPointerException anyway. I'll add a serialization test where `processorTag` is null though.

Reply:
Digging through the history, it looks like `writeTo` has been broken for 7 years -- https://github.com/elastic/elasticsearch/commit/9f48df9736655af2038a41af4f210e9336121f9e#diff-d67f5b3d1847a6f93411ddaf948733f2454d8fe11fb3851335133c830901853e -- there in `toXContent` we check whether `processorTag` is null (so it clearly can be null at that point) but not in `writeTo`. I'm guessing we've never actually hit this code in a real use before now. So I could back out that version check.